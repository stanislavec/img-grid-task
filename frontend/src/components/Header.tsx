import React from 'react';
import ButtonLink from './ButtonLink';
import scss from './Header.module.scss';

function Header() {
  return (
    <header className={scss.header}>
      <ButtonLink to="/">Photos</ButtonLink>
      <ButtonLink to="/favorites">Favorites</ButtonLink>
    </header>
  );
}

export default Header;
