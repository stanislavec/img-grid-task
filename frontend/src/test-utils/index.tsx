import React, { ReactElement } from 'react';
import { createBrowserRouter, createMemoryRouter, RouterProvider } from 'react-router-dom';

export const withRouter = (Component: ReactElement<unknown, any>, path: string = '/') => {
  const router = createBrowserRouter([
    {
      path,
      element: Component,
    },
  ]);

  return <RouterProvider router={router} />;
};

export const withMemotyRouter = (
  Component: ReactElement<unknown, any>,
  { route = '/', path = '/' } = {},
) => {
  const router = createMemoryRouter(
    [
      {
        path,
        element: Component,
      },
    ],
    {
      initialEntries: [route],
    },
  );
  return <RouterProvider router={router} />;
};
