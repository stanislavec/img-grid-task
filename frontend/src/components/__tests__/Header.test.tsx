import { render } from '@testing-library/react';
import { withRouter } from '../../test-utils';
import Header from '../Header';

describe('Header', () => {
  it('is defined', () => {
    const elem = render(withRouter(<Header />));
    expect(elem.baseElement).toBeDefined();
  });

  it('throws error with no required provider exists', () => {
    try {
      const elem = render(<Header />);
      expect(elem.baseElement).toBeDefined();
    } catch (error) {
      expect(error).toBeInstanceOf(Error);
    }
  });

  it('matches snapshot', () => {
    const elem = render(withRouter(<Header />));
    expect(elem.container).toMatchSnapshot();
  });
});
