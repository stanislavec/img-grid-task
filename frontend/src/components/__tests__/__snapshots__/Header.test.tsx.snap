// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <a
      class="button active"
      href="/"
    >
      Photos
    </a>
    <a
      class="button"
      href="/favorites"
    >
      Favorites
    </a>
  </header>
</div>
`;
